### ISSUE TITLE
[Feature]: Expose aria-current information in accessibility snapshots

### ISSUE THREAD

**Issue by bennypowers** [microsoft/playwright#36913]
### 🚀 Feature Request

We make extensive use of `page.accessibility.snapshot` in https://github.com/redhat-ux/red-hat-design-system/. I noticed today that snapshots do not expose aria current. Please consider adding support.

### Example

```html
<a href="#">1</a>
<a href="#" aria-current="step">2</a>
```

```ts
console.log(await page.accessibility.snapshot())
```

Expected:
```json
{
  role: 'WebArea',
  name: '',
  children: [
    { role: 'link', name: '1' },
    { role: 'link', name: '2', current: 'step' },
    { role: 'link', name: '3' },
    { role: 'link', name: '4' },
    { role: 'link', name: '5' }
  ]
}
```

Actual:
```json
{
  role: 'WebArea',
  name: '',
  children: [
    { role: 'link', name: '1' },
    { role: 'link', name: '2' },
    { role: 'link', name: '3' },
    { role: 'link', name: '4' },
    { role: 'link', name: '5' }
  ]
}
```

### Motivation

We use a11y snapshots for semantic component testing. This is different than pass/fail whole page audits, and it gives us rich insights into our design system's semantics and AT behaviours.

**Comment by agg23** [comment]
The `accessibility.snapshot()` API has been deprecated for a while now. We recommend you use [Axe](https://www.deque.com/axe/) for accessibility testing.

**Comment by bennypowers** [comment]
I hope you'll reconsider. The [issue which deprecates the accessibility snapshot api](https://github.com/microsoft/playwright/issues/16159) has a number of comments that demonstrate
- that there is demand for this feature
- that axe is insufficient and doesn't cover the use case desired 

As well, there are a number of other open issues for this api, which also demonstrates the demand for this feature.
